comptoir-ejb/src/main/java/be/valuya/comptoir/service/AccountingUtils.java

```diff
@@ -63,14 +63,31 @@ public static BigDecimal calcEffectivePriceWithoutTaxes(Price price) {
         return effectiveVatExclusive;
     }
 
+     public static SalePrice calcSaleDiscount(SalePrice salePrice, Sale sale) {
+        BigDecimal discountRatio = Optional.ofNullable(sale.getDiscountRatio()).orElse(BigDecimal.ZERO).setScale(2, RoundingMode.HALF_UP);
+        BigDecimal base = salePrice.getBase();
+        BigDecimal taxes = salePrice.getTaxes();
+        
+        BigDecimal discountBaseAmount = base.multiply(discountRatio).setScale(2, RoundingMode.HALF_UP);
+        BigDecimal discountTaxesAmount = taxes.multiply(discountRatio).setScale(2, RoundingMode.HALF_UP);
```

Comment:
Ok for now, but we'll need to check when/how to round taxes (after total?). Did it many times in other projects, so I'll check that.